internal/backend/remote-state/kubernetes/backend.go

```diff
@@ -49,7 +51,22 @@ func New() backend.Backend {
 			"secret_suffix": {
 				Type:        schema.TypeString,
 				Required:    true,
-				Description: "Suffix used when creating the secret. The secret will be named in the format: `tfstate-{workspace}-{secret_suffix}`.",
+				Description: "Suffix used when creating the secret. The secret will be named in the format: `tfstate-{workspace}-{secret_suffix}`. Note that the backend may append its own numeric index to the secret name when chunking large state files into multiple secrets. In this case, there will be multiple secrets named in the format: `tfstate-{workspace}-{secret_suffix}-{index}`.",
+				ValidateFunc: func(v interface{}, s string) ([]string, []error) {
+					value := v.(string)
+					// Split the suffix by '-' and check the last segment.
+					parts := strings.Split(value, "-")
+					if len(parts) > 1 {
+						lastPart := parts[len(parts)-1]
```

Comment:
Here's a neat trick, you can use `strings.LastIndex` here to make this code a bit more concise:

```lastPart := value[strings.LastIndex(value, "-")+1:]```

This will always return the last segment of a string delimited with `-`, and if there's no `-` then it will return the whole string – which will also be invalid if it's an integer.

Reply:
Good one, this helped in making the code cleaner.